frontend/js/src/playlists/components/ImportPlaylistModal.tsx

```diff
@@ -0,0 +1,204 @@
+import * as React from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import { toast } from "react-toastify";
+
+import GlobalAppContext from "../../utils/GlobalAppContext";
+import { ToastMsg } from "../../notifications/Notifications";
+
+type ImportPlaylistModalProps = {
+  playlist?: JSPFPlaylist;
+  initialTracks?: JSPFTrack[];
+};
+
+export default NiceModal.create((props: ImportPlaylistModalProps) => {
```

Comment:
I think we don't need any props for this component, you can remove the ImportPlaylistModalProps type above completely.
```suggestion
export default NiceModal.create() => {
```